tutorial room 4 readme (#1218)
playmint/ds: contracts/src/maps/tutorial-room-4/README.md

# Downstream Game Creation Tutorial 4

## Aim
We will follow the steps below to learn about 
- Changing tile colours
- Changing unit model
- Billboards
- Custom plugin UI

Once complete, your map should something look like this:

<img src="./readme-images/step0.png" width=600>

> # 1. Setup

Reviewer: This heading and all subsequent ones should be H2 (only one H1 per page)

```suggestion
## 1. Setup
```

Follow steps 1 through 4 of `tutorial-room-1` to get started. This will walk you through:
- Deploying the game
- Spawning a unit

> - Creating a map *

Reviewer: ```suggestion
- Creating a map
```

- Deploying new tiles

https://github.com/playmint/ds/blob/main/contracts/src/maps/tutorial-room-1/README.md

\* The only change we want to make to these steps is in _Creating a map_. As this is the next room in the tutorial series, we will be creating our map at a different angle so it doesn't overlap with another tutorial room.

<img src="./readme-images/step1.png" width=600>

## Let's create the files
If you're following along, you should copy the `.yaml` files from `tutorial-room-4` into your own map. If you'd like to learn more about how we use the `.yaml` files, please review [tutorial-room-1](https://github.com/playmint/ds/blob/main/contracts/src/maps/tutorial-room-1/README.md)

> # 2. Changing Tile Colours
> ### Create file: DiscoCentre.js

Reviewer: ```suggestion
# 2. Changing Tile Colours

### Create file: DiscoCentre.js
```

Let's work on the `"Disco Centre"` we defined in the `.yaml`.

> To start with, make a `boolean` variable that we'll use to allow the player to change the behaviour of the tiles:
> ```js

Reviewer: We should try to keep newlines between blocks so as not to upset the markdown parsers

```suggestion
To start with, make a `boolean` variable that we'll use to allow the player to change the behaviour of the tiles:

```js
```

    ];
```

> - For the ID, we're using the billboard we found in the world.
> - For the value, we passing in the image URL.
>   - Remember to consider CORS security requirements for the images you're using.
> 

Reviewer: I think we can drop the CORS bit, and un-list this... I think including it would require too much explanation.

```suggestion
For the ID, we're using the billboard we found in the world.

For the value, we passing in the image URL. 
```

We'll also define `buttons` as a `const` array of `map`(s):
```js
    const buttons = [
        {
            text: `Change Billboard Image 🔄`,
            type: 'action',
            action: changeImg,
            disabled: false,
        },
    ];
```

The `action` is calling the function we created to iterate through the image URL array.

> In the `update` `return` block, be sure to set the map, and buttons to the const's we created.
> 
> - map
>     ```js
>     map: map,
>     ```
> 
> - buttons
>     ```js
>     buttons: buttons,
>     ```
> 

Reviewer: this is kinda hard to read, bit more of the surrounding context might help:

````suggestion
be sure to include these `map`, and `buttons` properties your update function `return`:

```js
return {
  version: 1,
  map: map,
  buttons: buttons,
  components: [ ... ]
  ...
};
```

````
